runtime/executor/src/main/java/org/apache/nemo/runtime/executor/data/BlockManagerWorker.java

```diff
@@ -203,11 +224,17 @@ public Block createBlock(final String blockId,
           "Block " + blockIdWildcard + " location unknown: "
             + "The block state is " + blockLocationInfoMsg.getState()));
       }
+      LOG.info("readblock, msg. get block id {}", blockLocationInfoMsg.getBlockId());
 
       // This is the executor id that we wanted to know
       final String blockId = blockLocationInfoMsg.getBlockId();
       final String targetExecutorId = blockLocationInfoMsg.getOwnerExecutorId();
-      final DataStoreProperty.Value blockStore = edgeProperties.get(DataStoreProperty.class).get();
+      DataStoreProperty.Value blockStore = edgeProperties.get(DataStoreProperty.class).get();
+      if (this.spilledBlocks.containsKey(blockId)) {
```

Comment:
Ditto. Shouldn't change ExecutionProperty values during runtime or should stop execution -> rewrite Physical -> resume execution